apiml/src/main/java/org/zowe/apiml/GatewayHealthIndicator.java

```diff
@@ -39,25 +42,27 @@
 @RequiredArgsConstructor
 public class GatewayHealthIndicator extends AbstractHealthIndicator {
 
-    private static final ApimlLogger apimlLog = ApimlLogger.of(GatewayHealthIndicator.class, YamlMessageServiceInstance.getInstance());
     private final DiscoveryClient discoveryClient;
 
+    @InjectApimlLogger
+    private final ApimlLogger apimlLog = ApimlLogger.empty();
+
     @Value("${apiml.catalog.serviceId:}")
     private String apiCatalogServiceId;
 
     private AtomicBoolean discoveryAvailable = new AtomicBoolean(false);
     private AtomicBoolean zaasAvailable = new AtomicBoolean(false);
-
+    private AtomicBoolean catalogAvailable = new AtomicBoolean(false);
     private AtomicBoolean startedInformationPublished = new AtomicBoolean(false);
 
     @Override
     protected void doHealthCheck(Builder builder) throws Exception {
-        var anyCatalogIsAvailable = apiCatalogServiceId != null && !apiCatalogServiceId.isEmpty();
-        var apiCatalogUp = !this.discoveryClient.getInstances(apiCatalogServiceId).isEmpty();
+        var anyCatalogIsAvailable = StringUtils.isNotBlank(apiCatalogServiceId);
+        catalogAvailable.compareAndSet(false, anyCatalogIsAvailable && !this.discoveryClient.getInstances(apiCatalogServiceId).isEmpty());
```

Comment:
It is changing the behaviour, since once time is catalog ready, it is set to true forever. Shouldn't be

Reply:
Yes, but this is in modulith and it is prepared for when catalog is part of it, it won't be disabled, will it?